### PR TITLE
Mixin for grid controller added, manages the "deletion confirmation" …

### PR DESCRIPTION
**Pull request by leonardospina**
…workflow for the ItemDeleter component

### CODE REVIEW

File: app/controller/grid/ItemDeleterGridControllerMixin.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
+Ext.define('CpsiMapview.controller.grid.ItemDeleterGridControllerMixin', {
+    extend: 'Ext.Mixin',
+
+    onDeleteRowClick: function (tableView, rowIndex, colIndex, item, e, record, tableRow) {
+        var me = this;
+        // what to do for deleting
+        var doDelete = function () {
```
Comment (geographika) [inline app/controller/grid/ItemDeleterGridControllerMixin.js:7]: Not sure I understand the intent here? This function can't be overridden, but does nothing?

File: app/controller/grid/ItemDeleterGridControllerMixin.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                success: function (/*record, operation*/) {
+                    // do something if the delete succeeded
+                },
+                callback: function (/*record, operation, success*/) {
+                    // do something whether the delete succeeded or failed
+                }
+            });
+        };
+        // what to do for not deleting
+        var dontDelete = function () {
```
Comment (geographika) [inline app/controller/grid/ItemDeleterGridControllerMixin.js:21]: How is this configured? Shouldn't it be a property on the mixin at least? Also can use the https://docs.sencha.com/extjs/6.7.0/classic/Ext.html#property-emptyFn if required.

File: app/controller/grid/ItemDeleterGridControllerMixin.js [hunk 3]
```diff
@@ -1,44 +1,59 @@
...
 Ext.define('CpsiMapview.controller.grid.ItemDeleterGridControllerMixin', {
     extend: 'Ext.Mixin',
 
+    beforeDelete: function(config) {return true;},
+    onRowDeleteFail: Ext.emptyFn,
+    onRowDeleteSuccess: Ext.emptyFn,
+    onRowDeleteCallback: Ext.emptyFn,
+
+    // what to do for deleting
+    doDelete: function (record) {
```
Comment (geographika) [inline app/controller/grid/ItemDeleterGridControllerMixin.js:25]: Rename to onRowDelete?

File: app/controller/grid/ItemDeleterGridControllerMixin.js [hunk 4]
```diff
@@ -1,44 +1,59 @@
...
+        record.erase({
+            failure: this.onRowDeleteFail,
+            success: this.onRowDeleteSuccess,
+            callback: this.onRowDeleteCallback
+        });
+    },
+
+    // what to do for not deleting
+    dontDelete: Ext.emptyFn,
```
Comment (geographika) [inline app/controller/grid/ItemDeleterGridControllerMixin.js:34]: Rename to onRowCancelDelete?

File: app/controller/grid/ItemDeleterGridControllerMixin.js [hunk 5]
```diff
@@ -1,44 +1,59 @@
...
-                callback: function (/*record, operation, success*/) {
-                    // do something whether the delete succeeded or failed
-                }
-            });
-        };
-        // what to do for not deleting
-        var dontDelete = function () {
-        };
-        var beforeDeleteConfig = { avoidConfirmationRequest: item.avoidConfirmationRequest || false };
+        var beforeDeleteConfig = { avoidConfirmationRequest: item.avoidConfirmationRequest || false, record: record };
```
Comment (geographika) [inline app/controller/grid/ItemDeleterGridControllerMixin.js:38]: Is item.avoidConfirmationRequest an ExtJS property? If not where is this set?

File: app/controller/grid/ItemDeleterGridControllerMixin.js [hunk 6]
```diff
@@ -1,44 +1,59 @@
...
+ *  - onRowDeleteCallback   => do something whether the delete succeeded or failed
+ * 
+ *  - doDelete              => the actual deletion implementation, if the user confirms
+ *  - dontDelete            => in case we want to do something when the user doesn't confirim or beforeDelete returns false
+ */
+
 Ext.define('CpsiMapview.controller.grid.ItemDeleterGridControllerMixin', {
     extend: 'Ext.Mixin',
 
+    beforeDelete: function(config) {return true;},
```
Comment (geographika) [inline app/controller/grid/ItemDeleterGridControllerMixin.js:19]: rename to onRowBeforeDelete?

**Review by geographika — CHANGES_REQUESTED** [review]

File: app/controller/grid/ItemDeleterGridControllerMixin.js (hunk 5 above)
Comment (leonardospina) [reply app/controller/grid/ItemDeleterGridControllerMixin.js:38]: It's on the ItemDeleter component

File: app/view/fileupload/FileUploadWindowController.js [hunk 7]
```diff
@@ -62,19 +62,20 @@ Ext.define('CpsiMapview.view.fileupload.FileUploadWindowController', {
...
                     var newFiledata = {
-                        attachementId: '',
+                        attachementId: action.result.data.attachmentId,
                         name: name,
                         description: descrip,
                         extension: 'Extension',
                         fileName: fileName,
-                        fileSize: ''
+                        fileSize: '',
+                        lastUpdatedDateUtc: Date()
```
Comment (geographika) [inline app/view/fileupload/FileUploadWindowController.js:78]: Shouldn't these values be returned from the server?

Comment (leonardospina) [reply app/view/fileupload/FileUploadWindowController.js:78]: no the attachment service only returns the ID

**Pull request merged**